sopel/tools/time.py

```diff
@@ -58,16 +78,23 @@ def validate_timezone(zone):
     If the zone is not valid, :exc:`ValueError` will be raised.
 
     .. versionadded:: 6.0
+
+    .. versionchanged:: 8.0
+
+        If ``zone`` is ``None``, raises a :exc:`ValueError` as if it was an
+        empty string or an invalid timezone instead of returning ``None``.
+
     """
     if zone is None:
-        return None
+        raise ValueError('Invalid time zone.')
 
     zone = '/'.join(reversed(zone.split(', '))).replace(' ', '_')
     try:
         tz = pytz.timezone(zone)
     except pytz.exceptions.UnknownTimeZoneError:
         raise ValueError('Invalid time zone.')
-    return tz.zone
+
+    return cast(str, tz.zone)
```

Comment:
Necessary because `types-pytz` defines the `zone` property as `str | None`, I take it?

I'm sure they have a reason for not defining `zone: None` on the base class and separately defining `zone: str` on its children, but… I really don't like having to use `cast()` 😩

Reply:
Yes. Sometimes, you have to help `mypy` a little, especially when the library has shitty type representation.